### PR TITLE
Use asset price data to calculate current OP token price in native currency

### PR DESCRIPTION
**Pull request by jkadamczyk**
## What changed (plus any additional context for devs)

I implemented a way to get token data, concretely price data to display values in users native currency as selected in settings.

### CODE REVIEW

**Review by benisgold — APPROVED** [review]

File: src/screens/rewards/RewardsSheet.tsx
Comment (jinchung) [inline src/screens/rewards/RewardsSheet.tsx:45]: console

**Comment by jinchung** [comment]
I don't see where you're fetching the actual OP token price for generic assets - right now, I believe it's falling back to getting the price if the user has the OP token already in their wallet. We have a list of tokens we want generic asset prices for that we always request (regardless of whether a user has them or not) which you can add to: https://github.com/rainbow-me/rainbow/blob/develop/src/redux/explorer.ts#L324

File: src/screens/rewards/components/RewardsEarnings.tsx
Comment (jinchung) [inline src/screens/rewards/components/RewardsEarnings.tsx:72]: for the non-fallback USD path, you may be looking for `convertAmountAndPriceToNativeDisplay`: https://github.com/rainbow-me/rainbow/blob/develop/src/helpers/utilities.ts#L272

File: src/screens/rewards/components/RewardsStats.tsx
Comment (jinchung) [inline src/screens/rewards/components/RewardsStats.tsx:81]: same here: for the non-fallback USD path, you may be looking for `convertAmountAndPriceToNativeDisplay`: https://github.com/rainbow-me/rainbow/blob/develop/src/helpers/utilities.ts#L272

**Review by skylarbarrera — APPROVED** [review]

**Comment by jkadamczyk** [comment]
I merged other PRs into once since I had the approvals, now waiting for CI

**Pull request merged**
